File: Beginner/deadlySin/deadlySin.c
#include <stdio.h>
#include <assert.h>

int main(void){
	int T,X,Y,temp;

	scanf("%d",&T);
	assert(T>=1 && T<=100000);

	while(T>0){
		scanf("%d %d",&X,&Y);

		assert(X>=0 && X<=1000000000);
		assert(Y>=0 && Y<=1000000000);

		if(X==0){
			printf("%d\n",Y);
		}else if(Y==0){
			printf("%d\n",X);
		}else{
			if(Y<X){
				temp = Y;
				Y = X;
				X = temp;
			}

			while(Y%X!=0){
				temp = Y%X;
				Y = X;
				X = temp;
			}

			printf("%d\n",2*X);
		}

		T--;
	}
	
	return 0;
}

File: Beginner/deadlySin/deadlySin.java
//run as java -ea deadlySin
import java.util.Scanner;

public class deadlySin{
	public static void main(String[] args){
		int T,X,Y,temp;
		Scanner scan = new Scanner(System.in);

		T = scan.nextInt();
		assert(T>=1 && T<=100000) : "T must be 1 <= T <= 100000";

		while(T>0){
			X = scan.nextInt();
			Y = scan.nextInt();

			assert(X>=0 && X<=1000000000) : "X must be 0 <= X <= 10^9";
			assert(Y>=0 && Y<=1000000000) : "Y must be 0 <= Y <= 10^9";

			if(X==0){
				System.out.println(Y);
			}else if(Y==0){
				System.out.println(X);
			}else{
				if(Y<X){
					temp = Y;
					Y = X;
					X = temp;
				}

				while(Y%X!=0){
					temp = Y%X;
					Y = X;
					X = temp;
				}

				System.out.println(2*X);
			}

			T--;
		}
	}
}